src/imagecapturecore.cs

```diff
@@ -0,0 +1,789 @@
+		void DidRemoveItems (ICCameraDevice camera, ICCameraItem[] items);
+	
+		[Export ("cameraDevice:shouldGetThumbnailOfItem:")]
+		bool ShouldGetThumbnail (ICCameraDevice cameraDevice, ICCameraItem ofItem);
+	
+		[Export ("cameraDevice:shouldGetMetadataOfItem:")]
+		bool ShouldGetMetadata (ICCameraDevice cameraDevice, ICCameraItem ofItem);
+	}
+
+	interface IICCameraDeviceDownloadDelegate {}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject))]
+	interface ICCameraDeviceDownloadDelegate {
+
+		[Export ("didDownloadFile:error:options:contextInfo:")]
+		void DidDownloadFile (ICCameraFile file, [NullAllowed] NSError error, [NullAllowed] NSDictionary<NSString, NSObject> options, [NullAllowed] IntPtr contextInfo);
+	
+		[Export ("didReceiveDownloadProgressForFile:downloadedBytes:maxBytes:")]
+		void DidReceiveDownloadProgress (ICCameraFile file, long downloadedBytes, long maxBytes);
+	}
+	
+	[BaseType (typeof(ICDevice))]
+	[DisableDefaultCtor] // ICDeviceBrowser creates instances of this class
+	interface ICCameraDevice {
+
+		[Export ("batteryLevelAvailable")]
+		bool BatteryLevelAvailable { get; }
+	
+		[Export ("batteryLevel")]
+		nuint BatteryLevel { get; }
+	
+		[Export ("contentCatalogPercentCompleted")]
+		nuint ContentCatalogPercentCompleted { get; }
+	
+		[NullAllowed, Export ("contents")]
+		ICCameraItem[] Contents { get; }
+	
+		[NullAllowed, Export ("mediaFiles")]
+		ICCameraItem[] MediaFiles { get; }
+	
+		[Export ("timeOffset")]
+		double TimeOffset { get; }
+	
+		[Export ("isAccessRestrictedAppleDevice")]
+		bool IsAccessRestrictedAppleDevice { get; }
+	
+		[NullAllowed, Export ("mountPoint")]
+		string MountPoint { get; }
+	
+		[Export ("tetheredCaptureEnabled")]
+		bool TetheredCaptureEnabled { get; }
+	
+		[Export ("filesOfType:")]
+		[return: NullAllowed]
+		string[] GetFiles (string fileUTType);
+	
+		[Export ("requestSyncClock")]
+		void RequestSyncClock ();
+	
+		[Export ("requestEnableTethering")]
+		void RequestEnableTethering ();
+	
+		[Export ("requestDisableTethering")]
+		void RequestDisableTethering ();
+	
+		[Export ("requestTakePicture")]
+		void RequestTakePicture ();
+	
+		[Export ("requestDeleteFiles:")]
+		void RequestDeleteFiles (ICCameraItem[] files);
+	
+		[Export ("cancelDelete")]
+		void CancelDelete ();
+	
+		[Export ("requestDownloadFile:options:downloadDelegate:didDownloadSelector:contextInfo:")]
+		void RequestDownloadFile (ICCameraFile file, [NullAllowed] NSDictionary<NSString, NSObject> options, IICCameraDeviceDownloadDelegate downloadDelegate, Selector didDownloadSelector, [NullAllowed] IntPtr contextInfo);
+	
+		[Export ("cancelDownload")]
+		void CancelDownload ();
+	
+		[Export ("requestUploadFile:options:uploadDelegate:didUploadSelector:contextInfo:")]
+		void RequestUploadFile (NSUrl fileUrl, [NullAllowed] NSDictionary<NSString, NSObject> options, NSObject uploadDelegate, Selector didUploadSelector, [NullAllowed] IntPtr contextInfo);
```

Comment:
This is missing an `unsafe` no? I'm wondering if we can add manual bindings for `contextInfo`.

Reply:
IntPtrs are fine without unsafe, it's pointers that make it require. And obviously ok, since it compiles.

Reply:
We do have an unsafe with a similar signature, which explains the comment from @VincentDondain

Reply:
There is no need for `unsafe` here